lib/array.js

```diff
@@ -38,6 +38,63 @@ const map = (
   }
 };
 
+const DEFAULT_OPTIONS = { min: 5, percent: 0.7 };
+
+const asyncMap = (
+  // Non-blocking map
+  items, // array, incoming
+  fn, // function, (value, index) => {}
+  options, // object (optional), { min, percent }
+  done // function (optional), on done callback function(result)
+) => {
+  if (typeof options === 'function') {
+    done = options;
+    options = DEFAULT_OPTIONS;
+  }
+
+  if (!items.length) {
+    if (done) done([]);
+    return;
+  }
+
+  const min = options.min || DEFAULT_OPTIONS.min;
+  const percent = options.percent || DEFAULT_OPTIONS.percent;
+
+  let begin;
+  let sum = 0;
+  let count = 0;
+
+  const relation = percent / (1 - percent);
+  const result = new Array(items.length);
```

Comment:
I case there is no callback provided, it may be wise not to create an array, as it would be discarded anyway.

Though I have no strong feelings about this one.